### ISSUE TITLE
post-process-forwarder-errors container restarting loop (Broker: Offset out of range (broker 1001))

### ISSUE THREAD

**Issue by dmitriev-ivan** [getsentry/self-hosted#2629]
### Self-Hosted Version

23.11.2

### CPU Architecture

x86_64

### Docker Version

Docker version 24.0.5, build ced0996

### Docker Compose Version

Docker Compose version v2.20.2

### Steps to Reproduce

I just run Sentry and get a bunch of errors and endless container restarts

### Expected Result

There shouldn't be any mistakes, should the limits be increased somewhere?

### Actual Result

```
docker logs -f -n 200 sentry-self-hosted-post-process-forwarder-errors-1
```

```
09:22:21 [INFO] arroyo.processing.processor: New partitions assigned: {Partition(topic=Topic(name='events'), index=0): 11090427}
Traceback (most recent call last):
  File "/usr/local/lib/python3.8/site-packages/arroyo/processing/processor.py", line 294, in run
    self._run_once()
  File "/usr/local/lib/python3.8/site-packages/arroyo/processing/processor.py", line 356, in _run_once
    self.__message = self.__consumer.poll(timeout=1.0)
  File "/usr/local/lib/python3.8/site-packages/sentry/consumers/synchronized.py", line 245, in poll
    message = self.__consumer.poll(timeout)
  File "/usr/local/lib/python3.8/site-packages/arroyo/backends/kafka/consumer.py", line 414, in poll
    raise OffsetOutOfRange(str(error))
arroyo.errors.OffsetOutOfRange: KafkaError{code=_AUTO_OFFSET_RESET,val=-140,str="fetch failed due to requested offset not available on the broker: Broker: Offset out of range (broker 1001)"}
09:22:21 [ERROR] arroyo.processing.processor: Caught exception, shutting down...
09:22:21 [INFO] arroyo.processing.processor: Closing <sentry.consumers.synchronized.SynchronizedConsumer object at 0x7f0a67479430>...
09:22:21 [INFO] arroyo.processing.processor: Partitions to revoke: [Partition(topic=Topic(name='events'), index=0)]
09:22:21 [INFO] arroyo.processing.processor: Partition revocation complete.
09:22:21 [INFO] arroyo.processing.processor: Processor terminated
/usr/local/lib/python3.8/site-packages/memcache.py:1303: SyntaxWarning: "is" with a literal. Did you mean "=="?
  if key is '':
/usr/local/lib/python3.8/site-packages/memcache.py:1304: SyntaxWarning: "is" with a literal. Did you mean "=="?
  if key_extra_len is 0:
Traceback (most recent call last):
  File "/usr/local/bin/sentry", line 8, in <module>
    sys.exit(main())
  File "/usr/local/lib/python3.8/site-packages/sentry/runner/__init__.py", line 195, in main
    func(**kwargs)
  File "/usr/local/lib/python3.8/site-packages/click/core.py", line 1157, in __call__
    return self.main(*args, **kwargs)
  File "/usr/local/lib/python3.8/site-packages/click/core.py", line 1078, in main
    rv = self.invoke(ctx)
  File "/usr/local/lib/python3.8/site-packages/click/core.py", line 1688, in invoke
    return _process_result(sub_ctx.command.invoke(sub_ctx))
  File "/usr/local/lib/python3.8/site-packages/click/core.py", line 1688, in invoke
    return _process_result(sub_ctx.command.invoke(sub_ctx))
  File "/usr/local/lib/python3.8/site-packages/click/core.py", line 1434, in invoke
    return ctx.invoke(self.callback, **ctx.params)
  File "/usr/local/lib/python3.8/site-packages/click/core.py", line 783, in invoke
    return __callback(*args, **kwargs)
  File "/usr/local/lib/python3.8/site-packages/click/decorators.py", line 33, in new_func
    return f(get_current_context(), *args, **kwargs)
  File "/usr/local/lib/python3.8/site-packages/sentry/runner/decorators.py", line 69, in inner
    return ctx.invoke(f, *args, **kwargs)
  File "/usr/local/lib/python3.8/site-packages/click/core.py", line 783, in invoke
    return __callback(*args, **kwargs)
  File "/usr/local/lib/python3.8/site-packages/click/decorators.py", line 33, in new_func
    return f(get_current_context(), *args, **kwargs)
  File "/usr/local/lib/python3.8/site-packages/sentry/runner/decorators.py", line 29, in inner
    return ctx.invoke(f, *args, **kwargs)
  File "/usr/local/lib/python3.8/site-packages/click/core.py", line 783, in invoke
    return __callback(*args, **kwargs)
  File "/usr/local/lib/python3.8/site-packages/sentry/runner/commands/run.py", line 653, in basic_consumer
    run_processor_with_signals(processor)
  File "/usr/local/lib/python3.8/site-packages/sentry/utils/kafka.py", line 13, in run_processor_with_signals
    processor.run()
  File "/usr/local/lib/python3.8/site-packages/arroyo/processing/processor.py", line 294, in run
    self._run_once()
  File "/usr/local/lib/python3.8/site-packages/arroyo/processing/processor.py", line 356, in _run_once
    self.__message = self.__consumer.poll(timeout=1.0)
  File "/usr/local/lib/python3.8/site-packages/sentry/consumers/synchronized.py", line 245, in poll
    message = self.__consumer.poll(timeout)
  File "/usr/local/lib/python3.8/site-packages/arroyo/backends/kafka/consumer.py", line 414, in poll
    raise OffsetOutOfRange(str(error))
arroyo.errors.OffsetOutOfRange: KafkaError{code=_AUTO_OFFSET_RESET,val=-140,str="fetch failed due to requested offset not available on the broker: Broker: Offset out of range (broker 1001)"}
Updating certificates in /etc/ssl/certs...
0 added, 0 removed; done.
Running hooks in /etc/ca-certificates/update.d...
done.
```

```
docker logs -f -n 200 sentry-self-hosted-post-process-forwarder-transactions-1
```

```
/usr/local/lib/python3.8/site-packages/memcache.py:1303: SyntaxWarning: "is" with a literal. Did you mean "=="?
  if key is '':
/usr/local/lib/python3.8/site-packages/memcache.py:1304: SyntaxWarning: "is" with a literal. Did you mean "=="?
  if key_extra_len is 0:
Traceback (most recent call last):
  File "/usr/local/bin/sentry", line 8, in <module>
    sys.exit(main())
  File "/usr/local/lib/python3.8/site-packages/sentry/runner/__init__.py", line 195, in main
    func(**kwargs)
  File "/usr/local/lib/python3.8/site-packages/click/core.py", line 1157, in __call__
    return self.main(*args, **kwargs)
  File "/usr/local/lib/python3.8/site-packages/click/core.py", line 1078, in main
    rv = self.invoke(ctx)
  File "/usr/local/lib/python3.8/site-packages/click/core.py", line 1688, in invoke
    return _process_result(sub_ctx.command.invoke(sub_ctx))
  File "/usr/local/lib/python3.8/site-packages/click/core.py", line 1688, in invoke
    return _process_result(sub_ctx.command.invoke(sub_ctx))
  File "/usr/local/lib/python3.8/site-packages/click/core.py", line 1434, in invoke
    return ctx.invoke(self.callback, **ctx.params)
  File "/usr/local/lib/python3.8/site-packages/click/core.py", line 783, in invoke
    return __callback(*args, **kwargs)
  File "/usr/local/lib/python3.8/site-packages/click/decorators.py", line 33, in new_func
    return f(get_current_context(), *args, **kwargs)
  File "/usr/local/lib/python3.8/site-packages/sentry/runner/decorators.py", line 69, in inner
    return ctx.invoke(f, *args, **kwargs)
  File "/usr/local/lib/python3.8/site-packages/click/core.py", line 783, in invoke
    return __callback(*args, **kwargs)
  File "/usr/local/lib/python3.8/site-packages/click/decorators.py", line 33, in new_func
    return f(get_current_context(), *args, **kwargs)
  File "/usr/local/lib/python3.8/site-packages/sentry/runner/decorators.py", line 29, in inner
    return ctx.invoke(f, *args, **kwargs)
  File "/usr/local/lib/python3.8/site-packages/click/core.py", line 783, in invoke
    return __callback(*args, **kwargs)
  File "/usr/local/lib/python3.8/site-packages/sentry/runner/commands/run.py", line 653, in basic_consumer
    run_processor_with_signals(processor)
  File "/usr/local/lib/python3.8/site-packages/sentry/utils/kafka.py", line 13, in run_processor_with_signals
    processor.run()
  File "/usr/local/lib/python3.8/site-packages/arroyo/processing/processor.py", line 294, in run
    self._run_once()
  File "/usr/local/lib/python3.8/site-packages/arroyo/processing/processor.py", line 356, in _run_once
    self.__message = self.__consumer.poll(timeout=1.0)
  File "/usr/local/lib/python3.8/site-packages/sentry/consumers/synchronized.py", line 245, in poll
    message = self.__consumer.poll(timeout)
  File "/usr/local/lib/python3.8/site-packages/arroyo/backends/kafka/consumer.py", line 414, in poll
    raise OffsetOutOfRange(str(error))
arroyo.errors.OffsetOutOfRange: KafkaError{code=_AUTO_OFFSET_RESET,val=-140,str="fetch failed due to requested offset not available on the broker: Broker: Offset out of range (broker 1001)"}
Updating certificates in /etc/ssl/certs...
0 added, 0 removed; done.
Running hooks in /etc/ca-certificates/update.d...
done.
09:29:56 [INFO] arroyo.processing.processor: New partitions assigned: {Partition(topic=Topic(name='transactions'), index=0): 72952138}
Traceback (most recent call last):
  File "/usr/local/lib/python3.8/site-packages/arroyo/processing/processor.py", line 294, in run
    self._run_once()
  File "/usr/local/lib/python3.8/site-packages/arroyo/processing/processor.py", line 356, in _run_once
    self.__message = self.__consumer.poll(timeout=1.0)
  File "/usr/local/lib/python3.8/site-packages/sentry/consumers/synchronized.py", line 245, in poll
    message = self.__consumer.poll(timeout)
  File "/usr/local/lib/python3.8/site-packages/arroyo/backends/kafka/consumer.py", line 414, in poll
    raise OffsetOutOfRange(str(error))
arroyo.errors.OffsetOutOfRange: KafkaError{code=_AUTO_OFFSET_RESET,val=-140,str="fetch failed due to requested offset not available on the broker: Broker: Offset out of range (broker 1001)"}
09:29:56 [ERROR] arroyo.processing.processor: Caught exception, shutting down...
09:29:56 [INFO] arroyo.processing.processor: Closing <sentry.consumers.synchronized.SynchronizedConsumer object at 0x7f6c681cba90>...
09:29:56 [INFO] arroyo.processing.processor: Partitions to revoke: [Partition(topic=Topic(name='transactions'), index=0)]
09:29:56 [INFO] arroyo.processing.processor: Partition revocation complete.
09:29:56 [INFO] arroyo.processing.processor: Processor terminated
```

```
docker logs -n 200 -f sentry-self-hosted-billing-metrics-consumer-1
```

```
arroyo.errors.OffsetOutOfRange: KafkaError{code=_AUTO_OFFSET_RESET,val=-140,str="fetch failed due to requested offset not available on the broker: Broker: Offset out of range (broker 1001)"}
```

**Comment by hubertdeng123** [comment]
Seems to be similar as this issue here:
https://github.com/getsentry/self-hosted/issues/2358

Can you try some of the steps there?

**Comment by zKoz210** [comment]
`--no-strict-offset-reset `

I added this to all 3 containers, unfortunately I did not save the full log, but the error was similar. At the same time, I had 120 GB of RAM filled with Redis. I restarted all the containers again, deleted the Redis dump and the errors disappeared. That said, I lost all my data within 2 days of the error occurring, but that was the least of the problems. This helped me get everything back to working order.
